controllers/routing/routing_delete_resources.go

```diff
@@ -0,0 +1,75 @@
+package routing
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/opendatahub-io/odh-platform/pkg/metadata"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// HandleResourceDeletion handles the removal of dependent resources when the target resource is being deleted.
+func (r *PlatformRoutingReconciler) HandleResourceDeletion(ctx context.Context, sourceRes *unstructured.Unstructured) (ctrl.Result, error) {
+	exportModes, found := extractExportModes(sourceRes)
+	if !found {
+		r.log.Info("No export modes found, skipping deletion logic", "sourceRes", sourceRes)
+
+		return ctrl.Result{}, nil
+	}
+
+	r.log.Info("Handling deletion of dependent resources", "sourceRes", sourceRes)
+
+	for _, exportMode := range exportModes {
+		if err := r.deleteResourcesByLabels(ctx, sourceRes, metadata.ResourceGVKs(exportMode)); err != nil {
```

Comment:
I think this can be local ` routingResources(exportMode)` as they're only relevant for this functionality - perhaps move the whole content of `gvk.go` to a new file in this package and give it more domain-specific name.

Reply:
[72a9617](https://github.com/maistra/odh-platform/pull/39/commits/72a961743f9f3b6976e1dbc9a590e274185da5f0)